src/assets/js/dynamic-addon-cards.js

```diff
@@ -73,6 +72,12 @@
   };
 
   const updateAddonCard = async (card, { parsedUserAgent }) => {
+    const baseURLFromBody = document.querySelector('body').dataset.amoBaseUrl;
+    // We only want to override the baseURL for the dev environment.
+    const AMO_BASE_URL =
+      baseURLFromBody && baseURLFromBody === 'https://addons-dev.allizom.org'
+        ? baseURLFromBody
+        : 'https://addons.mozilla.org';
```

Comment:
Is this really needed? I am fine with it but, in theory, the Eleventy build step should already perform this check?

Reply:
Are you sure? The way I was understanding it is that the Eleventy build step creates the initial versions of the cards, which does include a version of this logic, but the code in `dynamic-addon-cards.js` runs in the browser to update the cards, and it picks up the `baseURL` from the `body`, which would still be the value for the specific environment. Without a check here, wouldn't we be using the stage API when we are on stage, which is what we didn't want?

Reply:
Right, so we should inject the correct value in the data attribute. Instead of using `markup.baseURL`, we should use `markup.baseApiUrl` probably.

Reply:
I changed a bunch of code to do this, which allowed the logic for dev to reside in one place. This is ready for another look.